File: algorithms/1_easy/C/02843_count_symmetric_integers/solution.py
class Solution:
    def countSymmetricIntegers(self, low: int, high: int) -> int:
        count = 0

        for i in range(low, high + 1):
            temp = i

            digits = []

            while temp > 0:
                digits.append(temp % 10)
                temp //= 10

            if (len(digits) & 1) == 1:
                continue

            half = len(digits) // 2
            left = sum(digits[0 : half])
            right = sum(digits[half:])

            if left == right:
                count += 1

        return count

File: algorithms/1_easy/C/02843_count_symmetric_integers/solution.rs
impl Solution {
    pub fn count_symmetric_integers(low: i32, high: i32) -> i32 {
        let mut digits: Vec<i32> = Vec::with_capacity(6);
        let mut count = 0_i32;

        for i in low..=high {
            let mut temp = i;

            digits.resize(0, 0);

            while temp > 0 {
                digits.push(temp % 10);
                temp /= 10;
            }

            if (digits.len() & 1) == 1 {
                continue;
            }

            let half = digits.len() / 2;
            let left = digits[0..half].iter().sum::<i32>();
            let right = digits[half..].iter().sum::<i32>();

            if left == right {
                count += 1;
            }
        }

        count
    }
}
